presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -1021,4 +1050,54 @@ default MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession
     {
         return new MaterializedViewFreshness(false);
     }
+
+    /**
+     * Redirects to another table or view which may or may not in the same catalog.
+     * <p>
+     * This method is called whenever the engine references a table or view name, including both accessing and creating one. Connectors are
+     * expected to return {@link Optional#empty()} if the redirection doesn't happen. The engine calls this method in an iterative manner,
+     * i.e., using the redirected name as the argument for the next call, until an {@link Optional#empty()} is returned. If a redirection
+     * does happen, the engine will access the redirected table or view name instead of the original one and raise a REDIRECTED_TABLE warning.
+     * </p>
+     * <p>
+     * Here are some caveats worth noting:
+     * <li>
+     *     If you choose to implement this method, you should consider implementing {@link #listTableColumnsWithRedirection}, {@link #getViewsWithRedirection},
+     *     {@link #listTablePrivilegesWithRedirection}, too. These methods are called to answer information schema queries. You should also
+     *     consider implementing {@link #redirectTableRename}, which is used to redirect rename targets.
```

Comment:
what happens if i implement `redirectTable` but do not implement `redirectTableRename`?

Reply:
It really depends. In the Hive-Iceberg case, implementing `redirectTable` and not `redirectTableRename` will cause renames across catalogs, which will fail. However, it can be fine for other cases.

Reply:
across catalogs or across schemas?
(renames across catalogs are not possible anyway, right?)